lib/shutil_custom/__init__.py

```diff
@@ -1,52 +1,36 @@
 import os
-import platform
 import stat
+from shutil import SpecialFileError, _samefile
 try:
-    from shutil import SpecialFileError, Error
-except:
-    from shutil import Error
-from shutil import _samefile
+    from shutil import SameFileError
+except ImportError:
+    from shutil import Error as SameFileError
 
 
-def copyfile_custom(src, dst):
-    """Copy data from src to dst"""
-    if _samefile(src, dst):
-        raise Error("`%s` and `%s` are the same file" % (src, dst))
+BUFFER_SIZE = 1024 * 1024 if os.name == 'nt' else 16 * 1024
+
 
-    for fn in [src, dst]:
+def copyfile_custom(src, dst):
+    """Copy data from src to dst."""
+    def special_file(fn):
         try:
             st = os.stat(fn)
         except OSError:
             # File most likely does not exist
             pass
         else:
-            # XXX What about other special files? (sockets, devices...)
             if stat.S_ISFIFO(st.st_mode):
-                try:
-                    raise SpecialFileError("`%s` is a named pipe" % fn)
-                except NameError:
-                    raise Error("`%s` is a named pipe" % fn)
+                raise SpecialFileError("{!r} is a named pipe".format(fn))
 
-    try:
-        # Windows
-        O_BINARY = os.O_BINARY
-    except:
-        O_BINARY = 0
+    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
```

Comment:
For consistency's sake - my point still stands.
We should be using `io.open` for Python 3 compatibility, same arguments.
For the record, `python-future` does that too.

I doesn't matter much in this context, however we decided we are going to phase out `python-future`, and we should be replacing all built-in `open` with `io.open` to make sure it's compatible with both major versions.

Reply:
Good point, changed.